src/compat/posix/time/strptime.c

```diff
@@ -0,0 +1,334 @@
+				return 0;
+			}
+			tm->tm_wday = date;
+			return n_leng;
+
+		case 'y':
+			if ((date > 99)) {
+				return 0;
+			}
+			if ((date > 68) && (date < 100)) {
+				tm->tm_year = date;
+			} else {
+				tm->tm_year = date + 100;
+			}
+			return n_leng;
+
+		case'Y':
+			if (n_leng > 4) {
+				return 0;
+			}
+			tm->tm_year = date - 1900;
+			return n_leng;
+
+		case'H':
+			if ((date > 23)) {
+				return 0;
+			}
+			tm->tm_hour = date;
+			return n_leng;
+
+		case'I':
+			if ((date > 12) || (date == 0)) {
+				return 0;
+			}
+			tm->tm_hour = date;
+			return n_leng;
+
+		case 'M':
+			if ((date > 59)) {
+				return 0;
+			}
+			tm->tm_min = date;
+			return n_leng;
+
+		case 'S':
+			if ((date > 60)) {
+				return 0;
+			}
+			tm->tm_sec = date;
+			return n_leng;
+
+		case 'j':
+			if ((date > 366) || (date == 0)) {
+				return 0;
+			}
+			tm->tm_yday = date - 1;
+			return n_leng;
+
+		case 'd':
+		case 'e':
+			if ((date > 31) || (date == 0)) {
+				return 0;
+			}
+			tm->tm_mday = date;
+			return n_leng;
+
+		case 'C':
+			if ((date > 99)) {
+				return 0;
+			}
+			tm->tm_year = date * 100 - 1900;
+			return n_leng;
+
+		case 'U':
+		case 'W':
+			if ((date > 53)) {
+				return 0;
+			}
+			return n_leng;
+
+		case 'n':
+		case 't':
+			if (s[0] == ' ') {
+				return 1;
+			}
+			return 0;
+
+		case 'R':
+		case 'T':
+			res_equi = def_format("H", &s[0], tm);
+			if (res_equi == 0) {
+				return 0;
+			}
+			if (s[res_equi] != ':' ) {
+				return 0;
+			}
+			res_equi = 0;
+			res_equi = def_format("M", &s[i + 1], tm);
+			if (res_equi == 0) {
+				return 0;
+			}
+			if (format[0] == 'R') {
+				return res_equi;
+			}
+			if (s[res_equi] != ':' ) {
+				return 0;
+			}
+			res_equi = 0;
+			res_equi = def_format("S", &s[i + 1], tm);
+			if (res_equi == 0) {
+				return 0;
+			}
+			return res_equi;
+
+		case 'D':
+			res_equi = def_format("m", &s[0], tm);
+			if (res_equi == 0) {
+				return 0;
+			}
+			if (s[res_equi] != '/' ) {
+				return 0;
+			}
+			res_equi = 0;
+			res_equi = def_format("d", &s[i + 1], tm);
+			if (res_equi == 0) {
+				return 0;
+			}
+			if (s[res_equi] != '/' ) {
+				return 0;
+			}
+			res_equi = 0;
+			res_equi = def_format("y", &s[i + 1], tm);
+			if (res_equi == 0) {
+				return 0;
+			}
+			return res_equi;
+
+		default:
+			return 0;
+	}
+}
+
+/**
+ * The strptime implementation doesn't support such format characters as %c, %p, %r, %U, %W, %x, %X.
+ * Also it doesn't support all alternative locale-dependent versions of date and time representation
+ * and all format characters of glibc.
+ */
+
+char *strptime(const char *s, const char *format, struct tm *tm) {
+
+	int count_s = 0;
+	int count_f = 0;
+	int new_count;
+
+	while (s[count_s] != '\0') {
+		if (format[count_f] != '%') {
+			new_count = cmp_before_percent(&s[count_s], &format[count_f]);
+			if (new_count == 0) {
+				return NULL;
+			}
+			count_f += new_count;
+			count_s += new_count;
+		} else {
+			if (s[count_s] == ' ') {
```

Comment:
This `if` is excessive (if condition is false, then loop will do nothing anyway).
Also you can use `isspace()` function to check if it's space/tab or other whitespace symbols

Reply:
I fixed it.
Thank you!